### ISSUE TITLE
Push notification customisation

### ISSUE THREAD

**Issue by apurcaroiu** [AppLozic/Applozic-iOS-SDK#547]
**Feature Request**: 
Hello,

I am trying to do some customization around push notification but it does not seem possible with the current SDK.

1) Allow to change the notification sound within the application. 
    I am able to set it initially during registration, but there is no API available to update it if the user   wants to change the sound from within the application after he he logged in.
2) Ability to use notification service extensions to manipulate content before display it to the user.

Are any of these supported or can be achieved at the moment ? If not, is there a chance you can add them in feature releases ?

Thanks

**Comment by Sunilkumarr** [comment]
Hello,
1. Allow to change the notification sound within the application.
I am able to set it initially during registration, but there is no API available to update it if the user wants to change the sound from within the application after he logged in.
--> We have a method but it's not exposed we might expose this method [link](https://github.com/AppLozic/Applozic-iOS-SDK/blob/master/sample-with-framework/Applozic/Applozic/Services/ALRegisterUserClientService.m#L239) in one of our releases

2. Ability to use notification service extensions to manipulate content before display it to the user.
--> Could you please explain how this can be used in your application why do you want to manipulate?

**Comment by apurcaroiu** [comment]
1. That would be great. Is there any endpoint or code we can manually add until next releases ?
    I did not see any endpoint in the api explorer.
2. For android version we have a push notification setting for 'show preview', to allow us manipulate the notification message(hide the preview). Basically we would like to  emulate the behaviour that 'show preview' has from the IOS system with a custom switch in our app. I was thinking that using a notification service extension would allow us to do this.

Thanks for your quick reply

**Comment by Sunilkumarr** [comment]
1. That would be great. Is there any endpoint or code we can manually add until next releases ?
I did not see any endpoint in the api explorer.
--> Will check with our backend team if we have any way to update the from platform API.
    Till then you can try copy-pasting this method code [link](https://github.com/AppLozic/Applozic-iOS-SDK/blob/master/sample-with-framework/Applozic/Applozic/Services/ALRegisterUserClientService.m#L239-L311) in your project file 

2. For android version we have a push notification setting for 'show preview', to allow us manipulate the notification message(hide the preview). Basically we would like to emulate the behaviour that 'show preview' has from the IOS system with a custom switch in our app. I was thinking that using a notification service extension would allow us to do this.
--> Can you share us the video or screenshot that you want to emulate the behavior to support@applozic.com

**Comment by apurcaroiu** [comment]
For point 2, similar behaviour as described below using an in app setting
https://www.youtube.com/watch?v=FlW9wNzgFKc

**Comment by Sunilkumarr** [comment]
1. Allow changing the notification sound within the application.
I am able to set it initially during registration, but there is no API available to update it if the user wants to change the sound from within the application after he logged in.
--> We have released a new version of Applozic 7.6.0 you can update it to use the method for updating the notificationSoundFileName

This is how can call to update the file name
```
    ALUser * user = [[ALUser alloc]init];
    user.notificationMode = [ALUserDefaultsHandler getNotificationMode];
    user.notificationSoundFileName = <notificationSoundFileName here>;

    [registerUserClientService updateUser:user withCompletion:^(ALRegistrationResponse *response, NSError *error) {

        if (response && response.isRegisteredSuccessfully) {
            /// Update the deatils Successfully
        } else {
            ///  Failed to update
        }

    }]; 
```
2. I request you to connect at support@applozic.com our product team will validate if we need to add the support for it or not.

**Comment by stevenphillips80** [comment]
My account is hacked by my old girlfriend. iOS hacked Fuck y?all. Everyone is fucking me around hereOn Tue, Jun 16, 2020 at 10:08 AM Mukesh Thawani <notifications@github.com> wrote:
Closed #547.

?You are receiving this because you are subscribed to this thread.Reply to this email directly, view it on GitHub, or unsubscribe.
